synapse/config/_base.py

```diff
@@ -201,6 +201,26 @@ def load_config(cls, description, argv):
 
         Returns: Config object.
         """
+        config_parser = cls.create_argument_parser(description)
+        obj, _ = cls.load_config_with_parser(config_parser, argv)
+
+        return obj
+
+    @classmethod
+    def create_argument_parser(cls, description):
```

Comment:
probably makes more sense to construct the ArgumentParser in the caller and pass it in, making this `add_arguments_to_parser`?

Reply:
what would happen if we moved both `create_argument_parser` and `load_config_with_parser` to a derived class of `HomeserverConfig`?

Reply:
I'm not sure that would really buy us very much tbh?